saleor/graphql/discount/filters.py

```diff
@@ -169,6 +174,7 @@ class PromotionWhere(MetadataWhereFilterBase):
         help_text="Filter promotions by start date.",
     )
     is_old_sale = BooleanWhereFilter(method="filter_is_old_sale")
+    type = ListObjectTypeFilter(input_class=PromotionTypeEnum, method="filter_type")
```

Comment:
We should allow providing operation here (`eq`, `oneOf`). You should use `OperationObjectTypeWhereFilter` instead of `ListObjectTypeFilter`, and you need to create a specific input type for this enum. Take a look at `AttributeTypeEnumFilterInput` or `MeasurementUnitsEnumFilterInput`.